src/robots/subsystems_manager.py

```diff
@@ -0,0 +1,146 @@
+        self._power:PowerModel = PowerModel()
+
+    def map_into_currents(self):
+        mapped = {
+            "current_draw_obc": True,
+            "current_draw_motor_controller": self.is_turned_on("MOTOR_CONTROLLER"),
+            "current_draw_neutron_spectrometer": self.is_turned_on("NEUTRON_SPECTROMETER"),
+            "current_draw_apxs": self.is_turned_on("APXS"),
+            "current_draw_camera": self.is_turned_on("CAMERA"),
+            "current_draw_radio": self.is_turned_on("RADIO"),
+            "current_draw_eps": True,
+        }
+
+        print(mapped)
+
+        return mapped
+    
+    def is_turned_on(self, electronic):
+        if self._electronics_power_state[electronic] == PowerState.ON:
+            return True
+        
+        return False
+
+    def calculate_rssi(self, robot_position):
+        self._radio.rover_position = robot_position
+        rssi = self._radio.rssi()
+
+        return rssi
+    
+    def calculate_temperature(self, robot_position, interval_s):
+        self._thermal.rover_position = robot_position
+        self._thermal.sun_position = self.SUN_POSITION
+        self._thermal.step(interval_s)
+        t = self._thermal.temperatures()
+
+        return t
+    
+    def calculate_power_status(self, robot_position, interval_s):
+        self._power.set_device_states(self.map_into_currents())
+        self._power.set_sun_position(self.SUN_POSITION)
+        self._power.set_rover_position(robot_position)
+        self._power.step(interval_s)
+        status = self._power.status()
+   
+        return status
+
+    def turn_on(self, electronics:str):
+        if electronics not in self._electronics_power_state:
+            print("Invalid electronics naming: ", electronics)
+            return
+
+        self._electronics_power_state[electronics] = PowerState.ON
+
+    def turn_off(self, electronics:str):
+        if electronics not in self._electronics_power_state:
+            print("Invalid electronics naming: ", electronics)
+            return
+
+        self._electronics_power_state[electronics] = PowerState.OFF
+
+    def get_electronics_state(self, electronics:str):
+        if electronics not in self._electronics_power_state:
+            print("Invalid electronics naming: ", electronics)
+            return
+
+        return self._electronics_power_state[electronics]
+    
+    def get_electronics_states(self):
+        return self._electronics_power_state
+
+    def deploy_solar(self):
+        self._solar_panel_state = SolarPanelState.DEPLOYED
+
+    def stow_solar(self):
+        self._solar_panel_state = SolarPanelState.STOWED
+
+    def get_solar_state(self):
+        return self._solar_panel_state
+    
+    def set_go_nogo_state(self, new_state:GoNogoState):
+        self._go_nogo_state = new_state
+
+    def get_go_nogo_state(self):
+        return self._go_nogo_state
+
+    def get_obc_state(self):
+        return self._obc_state
+    
+    def set_obc_to_camera(self):
```

Comment:
minor. is it cleaner to have a function for setting each state or juste the function for setting any state? 
i feel 2nd option but what do you think?